src/org/isf/utils/jobjects/CursorManager.java

```diff
@@ -23,15 +23,18 @@ private void cleanUp() {
     }
   }
   private void clearQueueOfInputEvents() {
-    EventQueue q = Toolkit.getDefaultToolkit().getSystemEventQueue();
-    clearLock.lock();
-    try {
-      ArrayList<AWTEvent> nonInputEvents = gatherNonInputEvents(q);
-      for (Iterator<AWTEvent> it = nonInputEvents.iterator(); it.hasNext();)
-        q.postEvent((AWTEvent)it.next());
-    }finally {
-      clearLock.unlock();
-    }
+    EventQueue q = Toolkit.getDefaultToolkit().getSystemEventQueue(); 
+    EventQueue parentQueue = ((WaitCursorEventQueue)q).getParentQueue();
+    try{
+        synchronized(parentQueue){
+          synchronized(q) {
+            ArrayList<AWTEvent> nonInputEvents = gatherNonInputEvents(q);
+            for (Iterator<AWTEvent> it = nonInputEvents.iterator(); it.hasNext();)
+              q.postEvent((AWTEvent)it.next());
+          }
+        }
+    }catch(Exception exp){}
+    finally{}
```

Comment:
Is there a need for empty try/catch/finally here? 
Also, as a minor suggestion, can we simplify the for loop to use for-each style, to make the code less verbose? e.g.:
```
 for (AWTEvent nonInputEvent : gatherNonInputEvents(q)) {
          q.postEvent(nonInputEvent);
        }
```

Reply:
Change done. Thanks.